pkgs/development/compilers/dotnet/patch-nupkgs.nix

```diff
@@ -34,6 +34,7 @@ in writeShellScriptBin "patch-nupkgs" (''
       if [ "$magic" = $'\177ELF' ]; then return 0; else return 1; fi
   }
   cd "$1"
+'' + lib.optionalString stdenv.isLinux ''
```

Comment:
This change looks obviously correct; do you know why this wasn't there before?

Reply:
For VMR, packages weren't being patched on darwin before, but 9.0 now needs the ilcompiler fix below.

I suppose this could actually use `stdenv.hostPlatform.isElf`.

Reply:
I think I'll leave this as-is for now.  `isElf` is something to consider for the future, but everything else is using `isLinux` and `isDarwin`.  I think realistically this can happen if/when we make dotnet work on BSD.